dev-python/orjson/orjson-3.8.4.ebuild

```diff
@@ -0,0 +1,80 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+CARGO_OPTIONAL=1
+CRATES="
+	ahash-0.8.2
+	arrayvec-0.7.2
+	associative-cache-1.0.1
+	autocfg-1.1.0
+	beef-0.5.2
+	bytecount-0.6.3
+	castaway-0.2.2
+	cc-1.0.78
+	cfg-if-1.0.0
+	chrono-0.4.23
+	compact_str-0.6.1
+	encoding_rs-0.8.31
+	itoa-1.0.5
+	itoap-1.0.1
+	libc-0.2.139
+	libm-0.1.4
+	num-integer-0.1.45
+	num-traits-0.2.15
+	once_cell-1.17.0
+	packed_simd_2-0.3.8
+	pyo3-build-config-0.17.3
+	pyo3-ffi-0.17.3
+	rustversion-1.0.11
+	ryu-1.0.12
+	serde-1.0.152
+	serde_json-1.0.91
+	simdutf8-0.1.4
+	smallvec-1.10.0
+	target-lexicon-0.12.5
+	version_check-0.9.4
+"
+
+DISTUTILS_USE_PEP517=maturin
+PYTHON_COMPAT=( python3_{9..11} )
+QA_FLAGS_IGNORED=".*"
+
+inherit cargo distutils-r1
+
+DESCRIPTION="Fast, correct Python JSON library supporting dataclasses, datetimes, and numpy"
+HOMEPAGE="https://github.com/ijl/orjson"
+SRC_URI="
+	https://github.com/ijl/orjson/archive/refs/tags/${PV}.tar.gz -> ${P}.gh.tar.gz
+	$(cargo_crate_uris ${CRATES})
+"
+
+LICENSE="
+	Apache-2.0 MIT
+	BSD
+	|| ( Apache-2.0 MIT )
+	|| ( Apache-2.0 Apache-2.0-with-LLVM-exceptions MIT )
+	|| ( Apache-2.0 Boost-1.0 )
+"
+KEYWORDS="~amd64"
+SLOT="0"
+IUSE="test"
+
+BDEPEND="
+	test? (
+		dev-python/arrow[${PYTHON_USEDEP}]
+		dev-python/orjson[${PYTHON_USEDEP}]
+		dev-python/pendulum[${PYTHON_USEDEP}]
+		dev-python/psutil[${PYTHON_USEDEP}]
+		dev-python/pytz[${PYTHON_USEDEP}]
+		$(python_gen_cond_dep 'dev-python/numpy[${PYTHON_USEDEP}]' python3_{8,9})
+		$(python_gen_cond_dep '>=dev-python/xxhash-1.4.3[${PYTHON_USEDEP}]' python3_8)
+	)
+"
+
+distutils_enable_tests pytest
+
+src_unpack() {
+	cargo_src_unpack
+}
```

Comment:
You don't need to redefine it, distutils-r1 doesn't override src_unpack.

Reply:
Without it I am getting a failure which is not caused by network AFAIK. Don't understand why.

`ebuild orjson/orjson-3.8.4.ebuild clean manifest test`:
```
>>> Compiling source in /var/tmp/portage/portage/dev-python/orjson-3.8.4/work/orjson-3.8.4 ...
 * python3_9: running distutils-r1_run_phase distutils-r1_python_compile
 *   Building the wheel for orjson-3.8.4 via maturin
gpep517 build-wheel --backend maturin --output-fd 3 --wheel-dir /var/tmp/portage/portage/dev-python/orjson-3.8.4/work/orjson-3.8.4-python3_9/wheel
2023-04-03 17:30:07,321 gpep517 INFO Building wheel via backend maturin
Running `maturin pep517 build-wheel -i /usr/bin/python3.9 --compatibility off --jobs=2 --skip-auditwheel`
    Updating crates.io index
warning: spurious network error (2 tries remaining): failed to resolve address for github.com: Temporary failure in name resolution; class=Net (12)
warning: spurious network error (1 tries remaining): failed to resolve address for github.com: Temporary failure in name resolution; class=Net (12)
error: Unable to update registry `crates-io`

Caused by:
  failed to fetch `https://github.com/rust-lang/crates.io-index`

Caused by:
  network failure seems to have happened
  if a proxy or similar is necessary `net.git-fetch-with-cli` may help here
  https://doc.rust-lang.org/cargo/reference/config.html#netgit-fetch-with-cli

Caused by:
  failed to resolve address for github.com: Temporary failure in name resolution; class=Net (12)
💥 maturin failed
  Caused by: Cargo metadata failed. Does your crate compile with `cargo build`?
  Caused by: `cargo metadata` exited with an error:
Error: command ['maturin', 'pep517', 'build-wheel', '-i', '/usr/bin/python3.9', '--compatibility', 'off', '--jobs=2', '--skip-auditwheel'] returned non-zero exit status 1
 ```

Reply:
Ok, I'll check it in a minute. Thanks for the update, I'll look at it soonish.